lib/progressive_hash_diff.rb

```diff
@@ -0,0 +1,30 @@
+class ProgressiveHashDiff
```

Comment:
I find the naming of this class a bit confusing, not sure what `Progressive` means in this context?